docs/en/observability/logs-troubleshooting.asciidoc

```diff
@@ -1,4 +1,133 @@
 [[logs-troubleshooting]]
 = Troubleshoot your logs
 
-_coming soon_
\ No newline at end of file
+This section provides possible solutions for errors you might encounter while onboarding your logs.
+
+[discrete]
+[[logs-troubleshooting-insufficient-priv]]
+== User does not have permissions to create API key
+
+If you don't have the required privileges to create an API key, you'll see the following error message:
+
+[source, plaintext]
+----
+User does not have permissions to create API key.
+
+Required cluster privileges are [`monitor`, `manage_own_api_key`] and 
+required index privileges are [`auto_configure`, `create_doc`] for 
+indices [`logs-*-*`, `metrics-*-*`], please add all required privileges 
+to the role of the authenticated user.
+----
+
+[discrete]
+[[logs-troubleshooting-insufficient-priv-solution]]
+=== Solution
+
+You need to either:
+
+* Have an administrator give you the `monitor` and `manage_own_api_key` cluster privileges and the `auto_configure` and `create_doc` indices privileges. Once you have these privileges, restart the onboarding flow.
+* Get an API key from an administrator and manually add the API to the {agent} configuration. See <<logs-stream-agent-config, Configure the {agent}>> for more on manually updating the configuration and adding the API key.
+
+[discrete]
+[[logs-troubleshooting-API-key-failed]]
+== Failed to create API key
+
+If you don't have the privileges to create `savedObjects` in {kib}, you'll see the following error message:
+
+[source, plaintext]
+----
+Failed to create API key
+
+Something went wrong: Unable to create observability-onboarding-state
```

Comment:
This is a comment for the devs, not docs: We should improve this error so users get already details in the error on what to do.